spinal_levels/README.md

```diff
@@ -0,0 +1,78 @@
+### PAM50 Levels
```

Comment:
## Sharp transition between level values (with no gradient/smoothness)

> * Is that normal that in the new version, values seem almost binary, ie: the same value is spread out along the S-I axis, without "smoothness" to it. Eg: zoomed version of C2_new (same value across ~20 voxels). Maybe we should inquired the Phillips group about that?

![image](https://user-images.githubusercontent.com/16181459/219680636-3efbe2c9-44b5-4574-aa43-01e81a6c7def.png)

Reply:
What I don't understand from this diagram, is that the "mean" is not a single value in the absissa, but a range. I'm not sure how to interpret that...  i'll answer in the email

Reply:
The email has been answered, and there has been some back and forth discussion as follows:

<details>
<summary> Email exchange </summary>

> > > **Julien Cohen-Adad**: What I don't understand from this diagram, is that the "mean" is not a single value, but a range. I'm not sure how to interpret that... To me, if the location of subject 1 is Z=[3,5], subject 2 [4,6] and subject 3 [5,7], then the "average" should be (one line corresponds to a value of Z):
> > > 
> > > ```
> > > 1:0
> > > 2:0
> > > 3: 0.33
> > > 4: 0.66
> > > 5: 1
> > > 6: 0.66
> > > 7: 0.33
> > > 8: 0
> > > etc.
> > > ```
> > > 
> > > So I would not expect to see such a "block" of ones.
> > > 
> > > Can you please provide us with more details about how this mean spinal segment was obtained?
> >
> > **Aaron Phillips**: The dataset that we used to estimate probabilities for each spinal cord level (from the Mendez paper) contained only the population mean and standard deviation across all 9 cadavers from which measurements were taken. As a result, it was not possible to create probability estimates for each layer (1 rostral-caudal voxel level) in the PAM50 model. 
> > 
> > We used the total length of the spinal cord between C2 and L5 in the PAM50 model, and with the measurements from the Mendez paper to determine the midpoint of each spinal cord level based on the length of each segment relative to the total length. With the data that we had, we moved forward by estimating the probability to be 1 for each spinal cord level, as it was given as a population mean, and added uncertainty based on the standard deviations. We thought that this approach was feasible given the limitations of the data that we used, relying on a mean length (±st-dev) for a population.  
> >
> > Happy to clarify further
>
> **Julien Cohen-Adad**: Thank you for your answers. 
> 
> I think I finally understand what you did, and why I was confused earlier. Our approach, with the PAM50, was to represent the mid-point of the nerve rootlets, whereas in your case (please correct me if I'm wrong), you represent the complete length of a spinal segment (ie: the entire "fanning" of rootlets entry point, which is not a single point in space, but a "range"). > So, if I understand correctly, you did the following:
> - compute the length of the PAM50 cord between C2 and L5 mid-point, let's call it "cord(PAM50)". Related question: Did you compute the total length between C2 and L5 mid-points, or between C2 (most rostral of nerve root entry) and L5 (most caudal of nerve root entry). I presume you did the former, because the PAM50 does not have information about most rostral/caudal entry of nerve rootlets.
> - got the length of the Mendez cords between C2 and L5 mid-point, averaged across subjects let's call it `cord(Mendez)`
> - got the length of each spinal segment (ie: from the most rostral to the most caudal nerve rootlet), averaged across subjects. Let's call it `length_segment[i](Mendez)` (i stands for a segment, eg: L5).
> - scale the length of each spinal segment to match the PAM50, using the following equation: 
>     - `length_segment[i](PAM50) = length_segment[i](Mendez) * cord(PAM50) / cord(Mendez)`
> - scale the position of the mid-points of each spinal level from the average Mendez cord, to the PAM50 cord, using the same normalization strategy as in the previous point.
> - scale the STD of each spinal level from the average Mendez cord, to the PAM50 cord, using the same normalization strategy. By the way: what STD are we talking about here? Is that the STD that represents the variation of the mid-point along the rostro-caudal axis?
> - build the PAM50 spinal level template, by centering each segment in the mid-points, and by adding, at the rostral and caudal edge of the segment, a half-gaussian function with height of one and STD of value calculated at the previous point.
> 
> Also, if my description above is correct, then I do not understand the discrepancy between the figure of the article, and the observed spinal segments. More specifically, the levels in the figure appear to vary more like a Gaussian along the rostro-caudal axis, while the generated spinal segments look more like a segment of "ones", with short smoothness at the edges:
> 
> ![image](https://user-images.githubusercontent.com/16181459/226619260-191d13d9-0d14-4aef-9be9-bc094420352e.png)
>
> Could you share the code and the data that were used to generate the spinal level atlas? That way we will be able to understand exactly what was done. In the future, if users ask us how these levels were calculated, we will be able to point to the exact code, which is better practice for transparency and reproducibility.
>
> About the data: Did you contact Mendez et al., or did you use the values they report in their article? I assume they report all results in the supplementary material, however the link to the supplementary material does not work in the PDF of the article. They also say that the suppl. material is available on the mayoclinic website (see screenshot below), but when going to this website and searching for their article, I do not see a link to the suppl. material. Can you please send it to us if you have it?
> 
> ![image](https://user-images.githubusercontent.com/16181459/226619688-e336988c-d5b8-4e79-95b2-fa8fe13a48bf.png)

</details>

We are now waiting on a response from Aaron to confirm the methods used, and to potentially supply us with the raw data + processing code. Until then, I'm not sure if there is much we can do to proceed.